android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/activity/ArchiveBuilderFactory.java

```diff
@@ -0,0 +1,80 @@
+package org.sagebionetworks.bridge.android.manager.activity;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.sagebionetworks.bridge.android.BridgeConfig;
+import org.sagebionetworks.bridge.android.data.Archive;
+import org.sagebionetworks.bridge.rest.model.ActivityType;
+import org.sagebionetworks.bridge.rest.model.SchemaReference;
+import org.sagebionetworks.bridge.rest.model.SurveyReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.sagebionetworks.bridge.rest.model.ActivityType.SURVEY;
+import static org.sagebionetworks.bridge.rest.model.ActivityType.TASK;
+
+/**
+ * Creates Archive.Builder instances from an activity identifier, popoulated with correct metadata
+ * to identify the format of the payload to Bridge.
+ */
+public class ArchiveBuilderFactory {
```

Comment:
I think you could move this class to the ArchiveBuild class and make an inner public static class called "Factory"